src/dbnode/client/reader_slice_of_slices_iterator.go

```diff
@@ -205,8 +205,10 @@ func (it *readerSliceOfSlicesIterator) Size() (int, error) {
 	return size, nil
 }
 
-func (it *readerSliceOfSlicesIterator) Rewind() {
-	it.resetIndex()
+func (it *readerSliceOfSlicesIterator) Clone() xio.ReaderSliceOfSlicesIterator {
+	clone := newReaderSliceOfSlicesIterator(it.segments, it.pool)
+	clone.idx = it.idx
```

Comment:
is this not pointing at the start now?

Reply:
I think this is making sure we start at the same index of the iterator we're cloning.